smurfs/src/reducers/index.js

```diff
@@ -1,3 +1,50 @@
+import {
+  FETCHING,
+  FETCHING_SUCCESS,
+  FETCHING_FAILURE,
+  ADDING,
+  ADDED,
+  ERROR} from '../actions'
+
+const initialState = {
+  smurfs: [],
+  fetchingSmurfs: false,
+  addingSmurf: false,
+  updatingSmurf: false,
+  deletingSmurf: false,
+  error: null
+}
+
+const smurfReducer = (state = initialState, action) => {
+  switch(action.type) {
+    case FETCHING:
+      return{...state, fetchingSmurfs: true};
+
+      case FETCHING_SUCCESS:
+        return{...state,
+        fetchingSmurfs: false,
+      smurfs: [...state.smurfs, ...action.payload]
+    };
+
+    case FETCHING_FAILURE:
+      return{
+        ...state, fetchingSmurfs: false, error: action.payload
+      };
+
+    case ADDED:
+      return {
+        ...state, smurfs: [...action.payload] }
```

Comment:
Here as well I'd suggest a console.log to your state after this action is performed, maybe using the redux-logger can make this easier since you are using it in your application.
I'd like to see how this dta looks after this.
The right way to go for me would be inside of your brackets having like this [...state.smurfs, action.payload], that is probably what you want.